packages/react-dom/src/__tests__/ReactFlightIntegration-test.js

```diff
@@ -0,0 +1,317 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+// Polyfills for test environment
+global.ReadableStream = require('@mattiasbuelens/web-streams-polyfill/ponyfill/es6').ReadableStream;
+global.TextDecoder = require('util').TextDecoder;
+
+// Don't wait before processing work on the server.
+// TODO: we can replace this with FlightServer.act().
+global.setImmediate = cb => cb();
+
+let act;
+let Stream;
+let React;
+let ReactDOM;
+let ReactFlightDOMServer;
+let ReactFlightDOMClient;
+
+describe('ReactFlightIntegration', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    act = require('react-dom/test-utils').act;
+    Stream = require('stream');
+    React = require('react');
+    ReactDOM = require('react-dom');
+    ReactFlightDOMServer = require('react-dom/unstable-flight-server');
+    ReactFlightDOMClient = require('react-dom/unstable-flight-client');
+  });
+
+  function getTestStream() {
+    let writable = new Stream.PassThrough();
+    let readable = new ReadableStream({
+      start(controller) {
+        writable.on('data', chunk => {
+          controller.enqueue(chunk);
+        });
+        writable.on('end', () => {
+          controller.close();
+        });
+      },
+    });
+    return {
+      writable,
+      readable,
+    };
+  }
+
+  it.experimental('should resolve the root', async () => {
+    let {Suspense} = React;
+
+    // Model
+    function Text({children}) {
+      return <span>{children}</span>;
+    }
+    function HTML() {
+      return (
+        <div>
+          <Text>hello</Text>
+          <Text>world</Text>
+        </div>
+      );
+    }
+    function RootModel() {
+      return {
+        html: <HTML />,
+      };
+    }
+
+    // View
+    function Message({result}) {
+      return <p dangerouslySetInnerHTML={{__html: result.model.html}} />;
+    }
+    function App({result}) {
+      return (
+        <Suspense fallback={<h1>Loading...</h1>}>
+          <Message result={result} />
+        </Suspense>
+      );
+    }
+
+    let {writable, readable} = getTestStream();
+    ReactFlightDOMServer.pipeToNodeWritable(<RootModel />, writable);
+    let result = ReactFlightDOMClient.readFromReadableStream(readable);
+
+    let container = document.createElement('div');
+    let root = ReactDOM.createRoot(container);
+    act(() => {
+      root.render(<App result={result} />);
+    });
+    expect(container.innerHTML).toBe('<h1>Loading...</h1>');
+
+    await act(async () => {});
```

Comment:
How does not doing anything help resolve it? Is it just because of a promise tick?

Does making the first one an async act avoid this?

Reply:
Yeah. If we make the first one async, we lost the opportunity to observe the initial loading state.

I guess I could resolve it manually to delay it.

Reply:
We have another test that verifies gradual loading states so I can make this one not observe it.

Reply:
There might not be an intermediate state *without* act because the scheduler would wait until after the tick and we end up delaying a bit (JND). So not sure it's even correct to test this. Certainly not an important detail.